src/views/dialogs/ReferralDialog.tsx

```diff
@@ -0,0 +1,162 @@
+import { useEffect } from 'react';
+
+import { styled } from 'twin.macro';
+
+import { AFFILIATES_FEE_DISCOUNT } from '@/constants/affiliates';
+import { ButtonSize } from '@/constants/buttons';
+import { DialogProps, ReferralDialogProps } from '@/constants/dialogs';
+import { LocalStorageKey } from '@/constants/localStorage';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useAccounts } from '@/hooks/useAccounts';
+import { useLocalStorage } from '@/hooks/useLocalStorage';
+import { useReferralAddress } from '@/hooks/useReferralAddress';
```

Comment:
[q] are we planning on reusing this functionality? or we just separated as a hook for cleanliness

Reply:
Mostly separated for cleanliness, wanted to have react-queries in their own hook mainly